### PR TITLE
React capstone project - Metrics webapp

### PR DESCRIPTION
**Pull request by username-forGithub**
- [ ] Use React documentation.
- [ ] Use React components.
- [ ] Use React props.
- [ ] Use React Router.
- [ ] Connect React and Redux.
- [ ] Handle events in a React app.
- [ ] Write integration tests with a React testing library.
- [ ] Use styles in a React app.
- [ ] Use React life cycle methods.
- [ ] Apply React best practices and language style guides in code.
- [ ] Use store, actions and reducers in React.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import Listing from './pages/Listing';
+import Single from './pages/Single';
+import './App.css';
+
+function App() {
+  return (
+    <Router>
+      <Routes>
+        <Route path="/" element={<Listing />} />
+        <Route path="/details/:Id" element={<Single />} />
```
Comment (ShoiraTa) [inline src/App.js:12]: - [ ] OPTIONAL: When I run `npm start`, it opens page  `http://localhost:3001/react-capstone` this page opens as an empty page, only after I delete  `react-capstone` from url, I see the content. I recommend to fix the url and make sure that the page is not blank when someone runs it from console.

File: src/__tests__/App.test.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import store from '../redux/configureStore';
+import App from '../App';
+import '@testing-library/jest-dom/extend-expect';
+
+describe('app routes', () => {
+  it('app elements', () => {
+    const aa = render(<Provider store={store}><App /></Provider>);
+    expect(aa).toMatchSnapshot();
+  });
+});
```
Comment (ShoiraTa) [inline src/__tests__/App.test.js:14]: - [ ] I can see you successfully added integration tests using React Testing Library. Please kindly add unit tests for several components of the app and some pure functions in Redux code.

**Review by ShoiraTa — CHANGES_REQUESTED** [review]
Hi @username-forGithub ,

:thumbsup: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.

**First let me to point out:**

- Great commit history! ✔️ 
- You passed all the Linters! ✔️ 
- Professional pull request! ✔️ 
- Application works as expected ✔️ 

Keep on coding, good job so far! :clap:

There are some issues that you still need to work on to go to the next project but you are almost there!



 ## Required Changes  :recycle: 

_Check the comments under the review._ 

***
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear by mentioning my Github username [@shoirata].


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

* * *
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/pages/Single.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+import { useSelector } from 'react-redux';
+// import { fetchStocks } from '../redux/slices/dataSlice';
+import Navbar from '../components/Navbar';
+import Subnav from '../components/Subnav';
+
+function Single() {
+  const { Id } = useParams();
+  // const dispatch = useDispatch();
+  // useEffect(() => {
+  //   dispatch(fetchStocks());
```
Comment (FrederickMih) [inline src/pages/Single.js:12]: - [Optional] Please do not push commented code to your repo. Comments render your codebase messy

**Review by FrederickMih — APPROVED** [review]
Hi @username-forGithub, 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights :green_circle: 

- Great job for following the design guidelines :heavy_check_mark: 


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

File: src/pages/Single.js (hunk 3 above)
Comment (username-forGithub) [reply src/pages/Single.js:12]: Thank you @FrederickMih

**Pull request merged**